tar/storage/storage_transaction.c

```diff
@@ -327,16 +327,26 @@ storage_transaction_start(NETPACKET_CONNECTION * NPC, uint64_t machinenum,
 		warn0("Sequence number mismatch: Run --fsck");
 		goto err0;
 	case 2:
-		/* We should only get this for a write transaction start. */
-		if (type != 0) {
+		/* What operation were we attempting? */
+		switch (type) {
+		case 0:
+			/* write. */
+			warn0("Cannot start write transaction: "
+			    "Account balance is not positive.");
+			warn0("Please add more money to your tarsnap account");
+			goto err0;
+		case 2:
+		case 3:
+			/* fsck. */
```

Comment:
at first I wrote
```
/* Write transaction. */
```
but then I had a problem with fsck.  Would it be
```
/* fsck transaction. */
```
or
```
/* Fsck transaction. */
```

The latter makes me cringe, but the former looks bad in context (of other comments having an initial capital letter).

Reply:
Maybe `/* --fsck transaction using write key. */` and likewise s/write/delete/ ?

Reply:
I take back the second part of that.  This code path should only be triggered when using the write keys (whether fscking or not).